File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int csum = nums[0], msum = nums[0];
        for(int i=1;i<(int)nums.size();i++){
            csum = max(csum+nums[i], nums[i]);
            msum = max(csum, msum);
        }
        return msum;
    }
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum= 0, ans= Integer.MIN_VALUE,n= nums.length;
        for(int i=0;i<n;i++){
            if(sum>=0) sum+= nums[i];
            else{
                sum= nums[i];
            }
            ans= Math.max(ans, sum);
        }
        return ans;
    }
}